### DISCUSSION TITLE
How to allow more clickable area for Radio and Checkboxes with border and padding

### DISCUSSION THREAD

**Discussion by mcmxcdev** [mantinedev/mantine#3021] [Q&A]
Our design team decided to have the `Radio` and `Checkbox` components with border. 

Currently as expected by HTML standards, only the `input` or `label` are clickable and change the selection, but we want it to do so on any click inside the bordered area:

https://user-images.githubusercontent.com/16797721/203042881-a7bd1142-2d35-43e6-b464-0385641ab44b.mp4

***

I found https://ui.mantine.dev/category/inputs#image-checkboxes and similar solutions, but they seem quite verbose and I think there might be an easier solution.

**Comment by rtivital** [comment]
Example provided on Mantine UI website is the current way of implementing this.

**Reply by mattaningram** [reply]
I see how it works in Mantine UI website, but I feel this would cause accessibility issues with a checkbox being inside a `button`, and on larger teams it will be hard to insure other devs know to use that since most will assume to just use `Checkbox` and add padding.

Would be nice if there were an option of the checkbox `input` being a child of the `label` so we can resize the label as needed and the whole thing is always clickable as expected by developers and users and still matches common accessibility patterns.

**Comment by lawrenceora** [comment]
Hi. Were you able to implement the same thing with Radio buttons? I can't seem to influence whether a Radio inside a Radio.Group gets checked or not. The Mantine website only have checkboxes as an example, which i was able to recreate.
